docs/developer-guide.md

```diff
@@ -21,39 +21,256 @@
+        #                HyperParameter(name: param-2, type: CATEGORICAL, list: cat1, cat2, cat3),
+        #                HyperParameter(name: param-3, type: DISCRETE, list: 3, 2, 6),
+        #                HyperParameter(name: param-4, type: DOUBLE, min: 1, max: 5, step: )]
+        #   )
+        search_space = HyperParameterSearchSpace.convert(request.experiment)
+        # Convert the trials in GRPC request to the trials in algorithm side.
+        # trials example:
+        #   [Trial(
+        #       assignment: [Assignment(name=param-1, value=2), 
+        #                    Assignment(name=param-2, value=cat1), 
+        #                    Assignment(name=param-3, value=2), 
+        #                    Assignment(name=param-4, value=3.44)],
+        #       target_metric: Metric(name="metric-2" value="5643"), 
+        #       additional_metrics: [Metric(name=metric-1, value=435), 
+        #                            Metric(name=metric-3, value=5643)],
+        #   Trial(
+        #       assignment: [Assignment(name=param-1, value=3),
+        #                    Assignment(name=param-2, value=cat2),
+        #                    Assignment(name=param-3, value=6),
+        #                    Assignment(name=param-4, value=4.44)],
+        #       target_metric: Metric(name="metric-2" value="3242"), 
+        #       additional_metrics: [Metric(name=metric=1, value=123), 
+        #                            Metric(name=metric-3, value=543)],
+        trials = Trial.convert(request.trials)
+        #--------------------------------------------------------------
+        # Your code here
+        # Implment the logic to generate new assignments for the given request number.
+        # For example, if request.request_number is 2, you should return:
+        # [
+        #   [Assignment(name=param-1, value=3), 
+        #    Assignment(name=param-2, value=cat2), 
+        #    Assignment(name=param-3, value=3), 
+        #    Assignment(name=param-4, value=3.22)
+        #   ],
+        #   [Assignment(name=param-1, value=4), 
+        #    Assignment(name=param-2, value=cat4), 
+        #    Assignment(name=param-3, value=2), 
+        #    Assignment(name=param-4, value=4.32)
+        #   ],
+        # ]
+        list_of_assignments = your_logic(search_space, trials, request.request_number)
+        #--------------------------------------------------------------
+        # Convert list_of_assignments to 
+        return api_pb2.GetSuggestionsReply(
+            trials=Assignment.generate(list_of_assignments)
+        )
 ```
 
-You can build all images from source for v1alpha2 as follows:
+### Make the algorithm a GRPC server
 
-```bash
-make images
+Create a package under [cmd/suggestion](../cmd/suggestion). Then create the main function and Dockerfile. Here is an example: [cmd/suggestion/hyperopt](../cmd/suggestion/hyperopt). Then build the Docker image.
```

Comment:
Please also highlight the service should serve with 6789 port

Reply:
SGTM

Reply:
@gaocegege Missed this?

Reply:
Oh yeah. I am adding it.